rest_api/sawtooth_rest_api/messaging.py

```diff
@@ -161,10 +182,21 @@ def __init__(self, socket, msg_router):
             content=message_content,
             message_type=message_type)
 
-        try:
-            await self._socket.send_multipart([message.SerializeToString()])
-        except asyncio.CancelledError:
-            raise
+        # Send the message. Backoff and retry in case of an error
+        backoff = _Backoff(max_retries=3,
+                           interval=200,
+                           error=SendBackoffTimeoutError())
+
+        while True:
+            try:
+                await self._socket.send_multipart(
+                    [message.SerializeToString()])
+                break
+            except asyncio.CancelledError:
+                try:
+                    await backoff.do_backoff()
```

Comment:
This back off should be done only when `zmq.error.Again` is thrown, not at a `asyncio.CancelledError`.  The latter is thrown when the loop is shutdown and the underlying future is canceled.